lib/src/rules/cast_nullable_to_non_nullable.dart

```diff
@@ -48,10 +48,13 @@ class CastNullableToNonNullable extends LintRule implements NodeLintRule {
         );
 
   @override
-  void registerNodeProcessors(NodeLintRegistry registry,
-      [LinterContext context]) {
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    if (!context.currentUnit.unit.featureSet.isEnabled(Feature.non_nullable)) {
```

Comment:
I added an `isEnabled` getter to the `context` to make this a bit more readable. Basically, you can drop `currentUnit.unit.featureSet`.

Reply:
Oh, that's great.  Thanks!